src/forms/toggleSwitch.test.jsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import { shallow, mount } from 'enzyme';
+
+const mockOnCallback = jest.fn();
+const mockOffCallback = jest.fn();
+
+import ToggleSwitch, {
+	TOGGLE_SWITCH_ACTIVE_CLASS,
+	TOGGLE_SWITCH_DISABLED_CLASS,
+	TOGGLE_SWITCH_CLASS,
+	// KNOB_CLASS,
+	KNOB_ACTIVE_CLASS,
+	KNOB_DISABLED_CLASS,
+	LABEL_CLASS,
+} from './ToggleSwitch';
+
+describe('ToggleSwitch', function() {
+	const DEFAULT_PROPS = {
+		name: 'testToggle'
+	};
+
+	const shallowWrapper = (props = {}) =>
+		shallow(<ToggleSwitch {...DEFAULT_PROPS} {...props} />);
+	const mountWrapper = (props = {}) =>
+		mount(<ToggleSwitch {...DEFAULT_PROPS} {...props} />);
+
+	describe('basic', function() {
+		it('renders inactive toggle switch', function() {
+			expect(shallowWrapper()).toMatchSnapshot();
+		});
+
+		it('renders checked toggle switch', function() {
+			expect(shallowWrapper({checked: true})).toMatchSnapshot();
+		});
+
+		it('has a label only if one is given', () => {
+			const component = mountWrapper({label: 'Turn it on?'});
+			expect(component.find(`.${LABEL_CLASS}`)).not.toBe(null);
```

Comment:
I was going to ask if it is more future proof to just look for the `label` tag rather than a class? `component.find('label')` but then I see we are not using a `label` element

Reply:
Will change this after doing your last suggestion